Libplanet.Net/Consensus/Context.cs

```diff
@@ -251,6 +253,70 @@ public void Dispose()
             return blockCommit;
         }
 
+        /// <summary>
+        /// Get <see cref="VotesRecall"/> containing <see cref="Vote"/>s
+        /// from <see cref="_heightVoteSet"/>  to respond <see cref="ConsensusBootstrapMsg"/>.
+        /// Round of returned <see cref="VotesRecall"/> is equal to <paramref name="round"/> + 1,
+        /// if <paramref name="round"/> is less than <see cref="Round"/>.
+        /// else, equal to <paramref name="round"/>.
+        /// </summary>
+        /// <param name="round">Round of requested <see cref="ConsensusBootstrapMsg"/>.</param>
+        /// <returns><see cref="VotesRecall"/> containing <see cref="Vote"/>s of proper
+        /// round.</returns>
+        public VotesRecall GetVotesRecall(int round)
+        {
+            ImmutableHashSet<Vote> votes;
+            if (round < Round)
+            {
+                round++;
+            }
+
+            votes = _heightVoteSet.PreVotes(round).List().Concat(
+                _heightVoteSet.PreCommits(round).List()).ToImmutableHashSet();
+
+            return new VotesRecallMetadata(
+                Height,
+                round,
+                DateTimeOffset.UtcNow,
+                _privateKey.PublicKey,
+                votes).Sign(_privateKey);
+        }
+
+        /// <summary>
+        /// Catch up <see cref="Round"/>s with given <paramref name="votesRecall"/>.
+        /// It only accepts <see cref="Vote"/>s those rounds are less than <see cref="Round"/> + 1.
+        /// </summary>
+        /// <param name="votesRecall">Received <see cref="VotesRecall"/> from peer
+        /// <see cref="Validator"/>.</param>
+        public void CatchupWithVotesRecall(VotesRecall votesRecall)
+        {
+            _logger.Debug(
+                "Catch up consensus with VotesRecall contains {count} votes : {votesRecall}",
+                votesRecall.Votes.Count,
+                votesRecall);
+            foreach (Vote vote in votesRecall.Votes)
+            {
+                if (vote.Round > Round + 1)
```

Comment:
What if consensus is being done in round higher than `Round + 1` ?

Reply:
`VotesRecall` is sent as a reply of `Bootstrap`, and `Votes` in `VotesRecall` are allowed to have only rounds which is same or +1 of requested `Bootstrap.Round`.
So, even though consensus is being done in round higher than `Round + 1`, bootstrapping peer will send lower round of `Bootstrap`, and corresponding `votesRecall` will be responded, and no `Vote`s are expected to be have round higher than `Round + 1`.
If boostrapping peer gives higher round of `Bootstrap`, we don't need to give him votes.

So, rejecting `Vote`s that are higher than `Round + 1` is intended. those votes will be added via subsequent `Bootstrap` and `VotesRecall`.